### ISSUE TITLE
Add unit tests - 100% coverage would be ideal.

### ISSUE THREAD

**Issue by jeffreydwalter** [jeffreydwalter/arlo#31]
Would be nice to have automated tests so we know the API is performing.

**Comment by Kerl1310** [comment]
Added some initial unit test code to get this underway. Happy to implement more over time if the initial implementation passes review

**Comment by jeffreydwalter** [comment]
@Kerl1310 thank you so much! I REALLY appreciate your contributions. Is there anything we need to do to get your tests hooked up to a github action?

**Comment by Kerl1310** [comment]
Theres a GitHub actions CI file included which _should_ just work once it's merged with any luck. If not I can fix it if there are any issues

**Comment by jeffreydwalter** [comment]
I merged your changes and pushed a new release commit, but didn't see any indication that the tests ran. Can you check in master and see if it's running?

**Comment by jeffreydwalter** [comment]
Looks like you set it up to only run on pull request? Am I reading this right?

**Comment by jeffreydwalter** [comment]
@jeffreydwalter I added support for running on push to master as well. Thanks again! I REALLY appreciate your contribution and am happy for anything else you feel like adding!

**Comment by Kerl1310** [comment]
Looks like the CI is running but it's having issues with the imports so the build is failing

**Comment by jeffreydwalter** [comment]
Yep! I was trying to figure it out, but don't really know how to fix it. If you have a few cycles to take a look I would be forever grateful!
